src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -34,10 +34,12 @@ DecisionTree<FitnessFunction,
                                         const size_t numClasses,
                                         const size_t minimumLeafSize)
 {
-  // Copy or move data.
-  typedef typename std::remove_reference<MatType>::type TrueMatType;
-  typedef typename std::remove_reference<LabelsType>::type TrueLabelsType;
+  using TrueMatType = typename std::remove_const<
```

Comment:
Could we use ``std::decay`` or maybe it makes sense to define a type alias since the expression is used multiple times? e.g.:

```
template<typename T>
using BaseType = typename std::remove_const<typename std::remove_reference<T>::type>::type;
```